rest/replicator_test.go

```diff
@@ -3932,7 +3927,25 @@ func TestSGR2TombstoneConflictHandling(t *testing.T) {
 		},
 	}
 
+	// requireTombstone validates tombstoned revision.
+	requireTombstone := func(t *testing.T, bucket *base.TestBucket, docID string) {
+		var rawBody db.Body
+		_, err := bucket.Get(docID, &rawBody)
+		if base.TestUseXattrs() {
+			require.True(t, base.IsDocNotFoundError(err))
+			require.Len(t, rawBody, 0)
+		} else {
+			require.NoError(t, err)
+			require.Len(t, rawBody, 1)
+			rawSyncData, ok := rawBody[base.SyncXattrName].(map[string]interface{})
+			require.True(t, ok)
+			_, ok = rawSyncData["tombstoned_at"].(float64)
```

Comment:
The `tombstoned_at` timestamp isn't the best way to check whether the document is a tombstone.  Checking the deleted flag in _sync.flags would be preferable.

Reply:
Added changes to use deleted flags from _sync.flags.